extensions/modules/widget-utils/widget-utils.lcb

```diff
@@ -0,0 +1,263 @@
+	      else
+	         delete the last char of pString
+	      end if
+	   end repeat
+	end if
+   
+   return pString
+end handler
+
+private handler colorComponentToString(in pComponent as Number)
+	return intToString(the rounded of (pComponent * 255))	
+end handler
+
+/*
+Summary: Converts a color to a string representing the color
+
+Parameters:
+pColor: A value of type com.livecode.canvas.Color
+pIncludeAlpha: Whether to include the alpha value in the string or not
+
+Example:
+property widgetColor 	get getColor
+metadata widgetColor.editor is "com.livecode.pi.colorwithalpha"
+
+private variable mColor as Color
+
+public handler OnSave(out rProperties as Array)
+	put colorToString(mColor, true) into rProperties["color with alpha"]
+end handler
+
+public handler getColor() returns String
+	-- the editor used for the widgetColor property expects the alpha value to be included
+	return colorToString(mColor, true)
+end handler
+
+Example:
+log colorToString(color [0.4,0.4,0.4,0.4])			-- logs "102,102,102,102"
+
+Description:
+Use the <colorToString> handler when logging colors, returning them to LiveCode script 
+via a property getter, or serialising them in the widget's stored properties array for saving.
+*/
+public handler colorToString(in pColor as Color, in pIncludeAlpha as Boolean) returns String
+	variable tComponents as List
+	variable tString as String
+	put [] into tComponents
+	push colorComponentToString(the red of pColor) onto tComponents
+	push colorComponentToString(the green of pColor) onto tComponents
+	push colorComponentToString(the blue of pColor) onto tComponents
+	
+	if pIncludeAlpha then
+		push colorComponentToString(the alpha of pColor) onto tComponents
+	end if
+	
+	combine tComponents with "," into tString
+	return tString
+end handler
+
+private handler stringToColorComponent(in pComponent as String)
+	return (pComponent parsed as number) / 255
+end handler
+
+/*
+Summary: Converts a string to a color
+
+Parameters:
+pColor: A comma delimited string representing a color in RGB / RGBA format
+
+Example:
+property widgetColor 	get getColor	set setColor
+metadata widgetColor.editor is "com.livecode.pi.colorwithalpha"
+
+private variable mColor as Color
+
+public handler OnLoad(in pProperties as Array)
+	setColor(pProperties["color"])
+end handler
+
+public handler setColor(in pColor as String)
+	put stringToColor(pColor) into mColor
+	redraw all
+end handler
+
+Description:
+Use the <stringToColor> handler when receiving colors from LiveCode script 
+via a property setter, or from the widget's stored properties array when loading.
+*/
+public handler stringToColor(in pString as String) returns Color
+	variable tComponentList as List
+	split pString by "," into tComponentList
+
+	variable tComponentCount
+	put the number of elements in tComponentList into tComponentCount
+	if tComponentCount is not 3 and tComponentCount is not 4 then
+		// Invalid number of components detected
+		throw "Invalid color"
+	end if
```

Comment:
This check isn't strictly necessary, as "color tColor" will throw an error if the list doesn't contain 3 or 4 numbers.
Sorry I missed this earlier.